tools/update-updated/index.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @fileoverview Task for updating `updated` YAML field.
+ *
+ * @author Matt Gaunt & Rob Dodson 💕
+ */
+
+// This task only runs against files in this directory.
+const contentDir = "src/site/content";
+
+const chalk = require("chalk");
+const fs = require("fs").promises;
+const moment = require("moment");
+const util = require("util");
+const exec = util.promisify(require("child_process").exec);
+const RE_UPDATED = /^updated:\s?(.*?)\n/m;
+const MSG_UPDATE = `Updated ${chalk.bold("updated")} in`;
+
+/**
+ * Gets the list of modified and renamed files that have been staged.
+ * This will ignore files that were added or deleted.
+ * @return {Promise<Array<string>>} Returns array of changed files.
+ */
+async function getChangedFiles() {
+  const cmd = `git diff --cached --name-only --diff-filter=MR`;
+  try {
+    const {stdout} = await exec(cmd, ".");
+    return stdout.split("\n");
+  } catch (err) {
+    throw err;
+  }
+}
+
+const run = async () => {
+  // List of all files that have changed
+  const changedFiles = await getChangedFiles();
+
+  for (const changedFile of changedFiles) {
+    if (changedFile.indexOf(contentDir) === -1) {
+      // File isn't a content file, skip it.
+      continue;
+    }
+
+    if (!changedFile.endsWith(".md")) {
+      // File isn't a Markdown file, skip it.
+      continue;
+    }
+
+    const fileContents = (await fs.readFile(changedFile)).toString();
+    const matched = RE_UPDATED.exec(fileContents);
+    if (!matched) {
+      // Updated YAML property is not in the file - nothing to do.
+      console.log(
+        `${chalk.black.bgYellow(
+          `Warning:`,
+        )} Could not find updated field in ${changedFile}.`,
+      );
+      continue;
+    }
+
+    const originalUpdated = matched[0];
+    const originalTimestamp = matched[1];
+    const momentNow = moment();
```

Comment:
Moment is how many kb and you pull it in purely to do.. what could be a string comparison? :(

```js
const now =  new Date().toISOString().substr(0, 10);
if (now < originalTimestamp) {
  continue;
}
```

Reply:
I'm mostly copying over what was in the webfu script which uses moment: https://github.com/google/WebFundamentals/blob/6cf700da9c2e9f5b3b3f271e27f2ad1e5b2e6a28/gulp-tasks/wfUpdatedOn.js#L71

I thought about removing it but chose to leave it for two reasons:
- I know folks shouldn't ship moment to the client, but this is running in node as part of the build process. The kb are irrelevant to me.
- I liked moment's interface for this and the formatting on line 71.